internal/pkg/deploy/cloudformation/stack/transformers.go

```diff
@@ -630,32 +630,124 @@ func convertTopic(t manifest.Topic, accountID, partition, region, app, env, svc
 	}, nil
 }
 
-func convertSubscribe(s *manifest.SubscribeConfig, validTopicARNs []string) (*template.SubscribeOpts, error) {
+func convertSubscribe(s *manifest.SubscribeConfig, validTopicARNs []string, accountID, region, app, env, svc string) (*template.SubscribeOpts, error) {
 	if s == nil || s.Topics == nil {
 		return nil, nil
 	}
 
+	sqsEndpoint, err := endpoints.DefaultResolver().EndpointFor(endpoints.SqsServiceID, region)
+	if err != nil {
+		return nil, err
+	}
+
 	var subscriptions template.SubscribeOpts
-	for _, sb := range *s.Topics {
-		ts, err := convertTopicSubscription(sb, validTopicARNs)
+	for _, sb := range s.Topics {
+		ts, err := convertTopicSubscription(sb, validTopicARNs, sqsEndpoint.URL, accountID, app, env, svc)
 		if err != nil {
 			return nil, err
 		}
 
 		subscriptions.Topics = append(subscriptions.Topics, ts)
 	}
+	queue, err := convertTopicQueue(s.Queue, sqsEndpoint.URL, accountID, app, env, svc)
+	if err != nil {
+		return nil, err
+	}
+	subscriptions.Queue = queue
 
 	return &subscriptions, nil
 }
 
-func convertTopicSubscription(t manifest.TopicSubscription, validTopicARNs []string) (*template.TopicSubscription, error) {
+func convertTopicSubscription(t manifest.TopicSubscription, validTopicARNs []string, url, accountID, app, env, svc string) (*template.TopicSubscription, error) {
 	err := validateTopicSubscription(t, validTopicARNs)
 	if err != nil {
 		return nil, fmt.Errorf(`invalid topic subscription "%s": %w`, t.Name, err)
 	}
+	queue, err := convertTopicQueue(t.Queue, url, accountID, app, env, svc)
+	if err != nil {
+		return nil, fmt.Errorf(`invalid topic subscription "%s": %w`, t.Name, err)
+	}
 
 	return &template.TopicSubscription{
 		Name:    aws.String(t.Name),
 		Service: aws.String(t.Service),
+		Queue:   queue,
+	}, nil
+}
+
+func convertTopicQueue(q *manifest.SQSQueue, url, accountID, app, env, svc string) (*template.SQSQueue, error) {
+	if q == nil {
+		return nil, nil
+	}
+	retention, err := convertTime(q.Retention, 0, 1209600)
+	if err != nil {
+		return nil, fmt.Errorf("invalid `retention`: %w", err)
+	}
+	delay, err := convertTime(q.Delay, 0, 900)
+	if err != nil {
+		return nil, fmt.Errorf("invalid `delay`: %w", err)
+	}
+	timeout, err := convertTime(q.Timeout, 0, 43200)
+	if err != nil {
+		return nil, fmt.Errorf("invalid `timeout`: %w", err)
+	}
+	deadletter, err := convertDeadLetter(q.DeadLetter)
+	if err != nil {
+		return nil, err
+	}
+
+	return &template.SQSQueue{
+		Name:       q.Name,
+		Retention:  retention,
+		Delay:      delay,
+		Timeout:    timeout,
+		KMS:        q.KMS,
+		DeadLetter: deadletter,
+		FIFO:       convertFIFO(q.FIFO),
+		AccountID:  accountID,
+		URL:        url,
+		App:        app,
+		Env:        env,
+		Svc:        svc,
+	}, nil
+}
+
+func convertTime(t *string, floor, ceiling float64) (*int, error) {
+	if t == nil {
+		return nil, nil
+	}
+
+	tm, err := time.ParseDuration(aws.StringValue(t))
```

Comment:
If we use the suggestion above, unmarshaling directly from the manifest as a `time.Duration`, we should be able to get rid of this call.